server/src/main/java/org/jboss/as/server/services/FunctionalVoidService.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.server.services;
+
+import java.util.function.Consumer;
+
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+import org.wildfly.common.function.ExceptionConsumer;
+
+/**
+ * Generic {@link Service} that provides no value, and delegates its start/stop to consumers.
+ * @author Paul Ferraro
+ */
+public class FunctionalVoidService implements Service<Void> {
```

Comment:
Do you intend for this to be used outside this module?

I ask because I'd like to break this module in two, roughly most things in the "deployment" package(s) in one module, usable by extensions, and then the rest in another, not usable by extensions. To be able to support end-user extension writing in EAP we're going to need to do that kind of separation. (Same in the contrller module.) So I want to avoid classes used by extensions getting spread around.